### PR TITLE
Introduce xsimd::batch_bool<T,A>::mask()

### LINKED ISSUE #753: Fast conversion between batch_bool and bit mask
We are recently using xsimd to make our [Velox](https://github.com/facebookincubator/velox) query evaluation engine portable.  One of the gap we found is that there is no convenient way to manipulate `batch_bool` efficiently.

What we do in our project is to convert them to bit mask, manipulate the bit mask, then convert it back to SIMD register.  To achieve fast conversion we need to build some global LUT.  You can refer to our implementation here: https://github.com/facebookincubator/velox/blob/main/velox/common/base/SimdUtil.h#L270-L297

If this aligns with your project's philosophy , we can help integrate the implementation into your project.

### PR DESCRIPTION
**Pull request by serge-sans-paille**
This method extracts an integer mask from a batch_bool, similar to _mm_move_mask*

Fix #753

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@Yuhta : I'd greatly appreciate a review of that patch. And if that's not asking too much, if you could contribute the arm implementation, that would be great ;-)

**Comment by Yuhta** [comment]
Thanks @serge-sans-paille , I am busy these days but will find some time to look at this

**Review by Yuhta — APPROVED** [review]

File: include/xsimd/arch/xsimd_avx.hpp [hunk 1]
```diff
@@ -620,6 +620,100 @@ namespace xsimd
             return _mm256_floor_pd(self);
         }
 
+        // from_mask
+        template <class A>
+        inline batch_bool<float, A> from_mask(batch_bool<float, A> const&, uint64_t mask, requires_arch<avx>) noexcept
```
[the remark this answers is not in the document]
Comment (serge-sans-paille) [reply include/xsimd/arch/xsimd_avx.hpp:625]: no :-/

**Comment by serge-sans-paille** [comment]
@Yuhta will you be able to contribute the arm version? cc @JohanMabille

**Pull request merged**